force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +120,19 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static Map<Id, Task> GetTasksDueNextWeek () {
+        Task[] Tasklist = [SELECT Id,Subject FROM Task WHERE ActivityDate <= :Date.today().addDays(7)];
+        Map<Id, Task> myTaskMap = new Map<Id, Task>();
+        if (Tasklist.size()>0) {
+            for (Task myT : Tasklist) {
+                myTaskMap.put(myT.Id, myT);
+            }
+            return myTaskMap;
+        }
+        else {
+            return null;
+        }
```

Comment:
Map has a special constructor you can use for SObject that will map they Id as the key to the associated SObject with that Id.
`new Map<Id, SObject>(aListOfSObjectsWithIds);`

Reply:
So you can pass it a query and it will take that and build the Id to SObject map for you.